[CI] S3BlobContainerRetriesTests.testReadRangeBlobWithRetries reproducibly fails

This looks like a newly added test and it has failed twice today. I was able to reproduce this failure locally in both `master` and `7.x`.

```
org.elasticsearch.repositories.s3.S3BlobContainerRetriesTests > testReadRangeBlobWithRetries FAILED
    java.lang.AssertionError: array lengths differed, expected.length=1 actual.length=0
        at __randomizedtesting.SeedInfo.seed([C3C28CF1ED038A75:16B862AC62B167BA]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.internal.ComparisonCriteria.assertArraysAreSameLength(ComparisonCriteria.java:76)
        at org.junit.internal.ComparisonCriteria.arrayEquals(ComparisonCriteria.java:37)
        at org.junit.Assert.internalArrayEquals(Assert.java:532)
        at org.junit.Assert.assertArrayEquals(Assert.java:341)
        at org.junit.Assert.assertArrayEquals(Assert.java:352)
        at org.elasticsearch.repositories.s3.S3BlobContainerRetriesTests.testReadRangeBlobWithRetries(S3BlobContainerRetriesTests.java:268)
REPRODUCE WITH: ./gradlew ':plugins:repository-s3:test' --tests "org.elasticsearch.repositories.s3.S3BlobContainerRetriesTests.testReadRangeBlobWithRetries" -Dtests.seed=C3C28CF1ED038A75 -Dtests.security.manager=true -Dtests.locale=sv-SE -Dtests.timezone=Europe/Sofia -Dcompiler.java=14
```

https://gradle-enterprise.elastic.co/s/zrusrq7cnhdve/tests/gharp3fr6cqlc-bfqjgyjdiiyka

--- a/plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobContainerRetriesTests.java
+++ b/plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobContainerRetriesTests.java
@@ -214,7 +214,6 @@ public class S3BlobContainerRetriesTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/54995")
     public void testReadRangeBlobWithRetries() throws Exception {
         final int maxRetries = randomInt(5);
         final CountDown countDown = new CountDown(maxRetries + 1);

--- a/plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobContainerRetriesTests.java
+++ b/plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobContainerRetriesTests.java
@@ -325,7 +325,6 @@ public class S3BlobContainerRetriesTests extends ESTestCase {
         assertThat(exception.getSuppressed().length, equalTo(0));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/54981")
     public void testReadBlobWithPrematureConnectionClose() {
         final int maxRetries = randomInt(20);
         final BlobContainer blobContainer = createBlobContainer(maxRetries, null, null, null);
@@ -547,13 +546,13 @@ public class S3BlobContainerRetriesTests extends ESTestCase {
         if (rangeEnd.isPresent()) {
             // adapt range end to be compliant to https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.35
             final int effectiveRangeEnd = Math.min(rangeEnd.get(), bytes.length - 1);
-            length = effectiveRangeEnd - rangeStart;
+            length = effectiveRangeEnd - rangeStart + 1;
         } else {
-            length = bytes.length - rangeStart - 1;
+            length = bytes.length - rangeStart;
         }
         exchange.getResponseHeaders().add("Content-Type", "text/plain; charset=utf-8");
         exchange.sendResponseHeaders(HttpStatus.SC_OK, length);
-        final int bytesToSend = length == 0 ? 0 : randomIntBetween(0, length - 1);
+        final int bytesToSend = randomIntBetween(0, length - 1);
         if (bytesToSend > 0) {
             exchange.getResponseBody().write(bytes, rangeStart, bytesToSend);
         }

